hyperparameter_search.py

```diff
@@ -394,19 +403,29 @@ def search_hyperparameters(model, no_of_epochs, no_of_trials, dataset, data_csv,
         device = torch.device('cuda')
     else:
         device = torch.device('cpu')
+
+    # Save the test dataset on disk to be loaded and used in train.py. This is to make sure the train set is kept unseend until the testing step in train.py.
+    # This is currently commented out if favor of using a seed given to random_split to achieve the same thing.
+    # model_class = get_model_class(MODEL_NAME, num_of_classes=NUM_CLASSES).to(device)
+    # id, model_name, timestamp = save_dataset_of_torch_model(model_class, test_dataset, "test_dataset")
+
     study = optuna.create_study(direction='minimize')
     study.optimize(func=lambda trial: objective(trial, MODEL_NAME, NUM_CLASSES, N_EPOCHS, OPTIMIZER_SEARCH_SPACE, \
         device, train_plant_dataloader, val_plant_dataloader, FLAG_EARLYSTOPPING, EARLYSTOPPING_PATIENCE),\
         n_trials=N_TRIALS)
 
-    ## TODO: how to save n best models/hyperparameter configurations, how to save the unseen test dataset,
-    ## which can then be used in train.py to evaluate the model.
-    # study.trials_dataframe()
-    # study.best_trial
-    # study.best_params
-    # study.best_trials
-    # torch.save(test_dataset, f"{MODEL_NAME}_test_dataset_with_train+valid_best_value_{study.best_value}.pt")
-    # torch.save(model.state_dict(), f'{MODEL_NAME}_weights_with_train+valid_best_value_{study.best_value}.pt')
+
+    df = study.trials_dataframe()
+    df = df.sort_values(by=['value'], ascending=True).iloc[0:9,:]
```

Comment:
This currently selects the 10 rows with the smallest value in the `value`-column. I'm not sure what values are in the column, just doublechecking that we want to select the 10 smallest and not the 10 largest (which could be achieved by changing to `ascending=False`)

Reply:
Fixed